### DISCUSSION TITLE
SWEET units that are missing from QUDT

### DISCUSSION THREAD

**Discussion by dr-shorthair** [qudt/qudt-public-repo#1456] [General]
The [SWEET ontology](https://github.com/ESIPFed/sweet) is another semantic resource that is widely used  in the earth and environment science community. Part of SWEET is a set of unit definitions, primarily in this graph: 

https://github.com/ESIPFed/sweet/blob/master/src/reprSciUnits.ttl

I've gone through them and matched most with QUDT concepts. Here are the ones for which there are **not** QUDT equivalents as far as I can find: 

This set are trivial permutations of things already in QUDT: 

- reprSciUnits:MYA
- reprSciUnits:ampereSquared
- reprSciUnits:century
- reprSciUnits:kelvinPerMeter
- reprSciUnits:kilogramRadianPerSecond
- reprSciUnits:meterSquaredPerSecondSquared
- reprSciUnits:millenium
- reprSciUnits:million_km2
- reprSciUnits:million_km3
- reprSciUnits:perAmpere
- reprSciUnits:perAmpereSquared
- reprSciUnits:perCandela
- reprSciUnits:perKilogram
- reprSciUnits:perKilogramMeterSquaredPerSecondKelvin
- reprSciUnits:perKilogramMole
- reprSciUnits:perRadian
- reprSciUnits:perSecondCubed
- reprSciUnits:perSecondSquared
- reprSciUnits:perSteradian
- reprSciUnits:perTesla
- reprSciUnits:perVolt
- reprSciUnits:siemensPerKilogram
- reprSciUnits:wattPerMeterSquaredPerSteradianPerWavelength

This one is interesting! (not sure if it is 3-month seasons - which is a cultural artefact)

- reprSciUnits:season

These are ratios and per-volume units that I'm not sure are managed in QUDT

- reprSciUnits:dimensionlessUnit
- reprSciUnits:ratio
- reprSciUnits:ppbv
- reprSciUnits:ppmv
- reprSciUnits:pptv
- reprSciUnits:volumeRatio

These I don't know about 

- reprSciUnits:FNU
- reprSciUnits:FTU
- reprSciUnits:FormazinTurbidityUnit
- reprSciUnits:normalizedUnit

In due course QUDT should be supplemented with (most of) these missing items, which are derived from actual usage in that community.

**Comment by jhodgesatmb** [comment]
I believe that the board decided to cross reference with SWEET back in
February.

On Thu, Sep 10, 2020 at 3:40 PM Simon Cox <notifications@github.com> wrote:

> The SWEET ontology <https://github.com/ESIPFed/sweet> is another semantic
> resource that is widely used in the earth and environment science
> community. Part of SWEET is a set of unit definitions, primarily in this
> graph:
>
> https://github.com/ESIPFed/sweet/blob/master/src/reprSciUnits.ttl
>
> I've gone through them and matched most with QUDT concepts. These are the
> ones for which there are *not* QUDT equivalents as far as I can find:
>
>    - reprSciUnits:FNU
>    - reprSciUnits:FTU
>    - reprSciUnits:FormazinTurbidityUnit
>    - reprSciUnits:MYA
>    - reprSciUnits:ampereSquared
>    - reprSciUnits:century
>    - reprSciUnits:dimensionlessUnit
>    - reprSciUnits:kelvinPerMeter
>    - reprSciUnits:kilogramRadianPerSecond
>    - reprSciUnits:meterSquaredPerSecondSquared
>    - reprSciUnits:millenium
>    - reprSciUnits:million_km2
>    - reprSciUnits:million_km3
>    - reprSciUnits:normalizedUnit
>    - reprSciUnits:perAmpere
>    - reprSciUnits:perAmpereSquared
>    - reprSciUnits:perCandela
>    - reprSciUnits:perKilogram
>    - reprSciUnits:perKilogramMeterSquaredPerSecondKelvin
>    - reprSciUnits:perKilogramMole
>    - reprSciUnits:perRadian
>    - reprSciUnits:perSecondCubed
>    - reprSciUnits:perSecondSquared
>    - reprSciUnits:perSteradian
>    - reprSciUnits:perTesla
>    - reprSciUnits:perVolt
>    - reprSciUnits:ppbv
>    - reprSciUnits:ppmv
>    - reprSciUnits:pptv
>    - reprSciUnits:ratio
>    - reprSciUnits:season
>    - reprSciUnits:siemensPerKilogram
>    - reprSciUnits:volumeRatio
>    - reprSciUnits:wattPerMeterSquaredPerSteradianPerWavelength
>
> In due course we should supplement the QUDT vocabulary with these missing
> items, which are derived from actual usage in that community.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/qudt/qudt-public-repo/issues/231>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AATQRWIJCSIMDLAWYQ3OKELSFFIUXANCNFSM4RGAZA7Q>
> .
>


-- 
Jack

**Comment by dr-shorthair** [comment]
The alignment should be done _from_ SWEET _to_ QUDT. 

I've done a first cut here  https://github.com/ESIPFed/sweet/blob/issue217-qudt-alignment/alignments/sweet-qudt-mapping.ttl which will be merged with the main branch of SWEET if https://github.com/ESIPFed/sweet/pull/221 is accepted. 
The list above is the set that need to be added to QUDT to allow a complete mapping from SWEET.

**Comment by jhodgesatmb** [comment]
That inclusion would have to follow the normal unit submission guidelines:

https://github.com/qudt/qudt-public-repo/wiki/Unit-Vocabulary-Submission-Guidelines

Especially considering that so little information is provided out of the
mandatory fields.

Jack

On Thu, Sep 10, 2020 at 3:40 PM Simon Cox <notifications@github.com> wrote:

> The SWEET ontology <https://github.com/ESIPFed/sweet> is another semantic
> resource that is widely used in the earth and environment science
> community. Part of SWEET is a set of unit definitions, primarily in this
> graph:
>
> https://github.com/ESIPFed/sweet/blob/master/src/reprSciUnits.ttl
>
> I've gone through them and matched most with QUDT concepts. These are the
> ones for which there are *not* QUDT equivalents as far as I can find:
>
>    - reprSciUnits:FNU
>    - reprSciUnits:FTU
>    - reprSciUnits:FormazinTurbidityUnit
>    - reprSciUnits:MYA
>    - reprSciUnits:ampereSquared
>    - reprSciUnits:century
>    - reprSciUnits:dimensionlessUnit
>    - reprSciUnits:kelvinPerMeter
>    - reprSciUnits:kilogramRadianPerSecond
>    - reprSciUnits:meterSquaredPerSecondSquared
>    - reprSciUnits:millenium
>    - reprSciUnits:million_km2
>    - reprSciUnits:million_km3
>    - reprSciUnits:normalizedUnit
>    - reprSciUnits:perAmpere
>    - reprSciUnits:perAmpereSquared
>    - reprSciUnits:perCandela
>    - reprSciUnits:perKilogram
>    - reprSciUnits:perKilogramMeterSquaredPerSecondKelvin
>    - reprSciUnits:perKilogramMole
>    - reprSciUnits:perRadian
>    - reprSciUnits:perSecondCubed
>    - reprSciUnits:perSecondSquared
>    - reprSciUnits:perSteradian
>    - reprSciUnits:perTesla
>    - reprSciUnits:perVolt
>    - reprSciUnits:ppbv
>    - reprSciUnits:ppmv
>    - reprSciUnits:pptv
>    - reprSciUnits:ratio
>    - reprSciUnits:season
>    - reprSciUnits:siemensPerKilogram
>    - reprSciUnits:volumeRatio
>    - reprSciUnits:wattPerMeterSquaredPerSteradianPerWavelength
>
> In due course we should supplement the QUDT vocabulary with these missing
> items, which are derived from actual usage in that community.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/qudt/qudt-public-repo/issues/231>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AATQRWIJCSIMDLAWYQ3OKELSFFIUXANCNFSM4RGAZA7Q>
> .
>


-- 
Jack

**Comment by dr-shorthair** [comment]
Yes - understood. I was just putting the list on record for the moment - I didn't expect it to be processed without more information (though most of them are trivial permutations of things already in QUDT.).
